src/insights_client/__init__.py

```diff
@@ -208,47 +230,62 @@ def run_phase(phase, client, validated_eggs):
     Call the run script for the given phase.  If the phase succeeds returns the
     index of the egg that succeeded to be used in the next phase.
     """
-    insights_command = [
-        sys.executable,
-        os.path.join(os.path.dirname(__file__), "run.py"),
-    ] + sys.argv[1:]
     config = client.get_conf()
 
-    all_eggs = [ENV_EGG, NEW_EGG] + validated_eggs
+    all_eggs = [NEW_EGG] + validated_eggs
+    if ENV_EGG is not None:
+        all_eggs = [ENV_EGG] + all_eggs
+        logger.debug("Environment egg defined as %s." % ENV_EGG)
+
+    for egg in all_eggs:
+        if config["gpg"]:
+            if not os.path.isfile(egg):
+                logger.debug("%s is not a file, can't GPG verify. Skipping.", egg)
+                continue
+            if not client.verify(egg)["gpg"]:
+                logger.debug("Not using egg: %s", egg)
+                continue
+        else:
+            logger.debug("GPG disabled by --no-gpg, not verifying %s.", egg)
 
-    for i, egg in enumerate(all_eggs):
-        if egg is None or (config["gpg"] and not os.path.isfile(egg)):
-            client_debug("Egg does not exist: %s" % egg)
-            continue
-        if config["gpg"] and not client.verify(egg)["gpg"]:
-            client_debug("WARNING: GPG verification failed. Not loading egg: %s" % egg)
-            continue
-        client_debug("phase '%s'; egg '%s'" % (phase["name"], egg))
+        logger.debug("phase '%s'; egg '%s'", phase["name"], egg)
 
         # prepare the environment
+        env = os.environ
+        logger.debug("Existing enviroment: %s", debug_environ(env))
+
         pythonpath = str(egg)
-        env_pythonpath = os.environ.get("PYTHONPATH", "")  # type: str
+        logger.debug("Adding to PYTHONPATH: %s", pythonpath)
+        env_pythonpath = env.get("PYTHONPATH", "")  # type: str
+
         if env_pythonpath:
             pythonpath += ":" + env_pythonpath
         insights_env = {
             "INSIGHTS_PHASE": str(phase["name"]),
             "PYTHONPATH": pythonpath,
         }
-        env = os.environ
+        logger.debug("Adding to the environment: %s", debug_environ(insights_env))
+
         env.update(insights_env)
 
+        current_path = os.path.dirname(__file__)
+        logger.debug("Current path: %s", current_path)
```

Comment:
Is this log statement telling us something useful?

Reply:
It prints the path of the _insights\_client/\_\_init\_\_.py_, which can be useful whenever the _\_\_init\_\_.py_ file is not run directly. Most commonly this is from the _insights-client_ command script, which, too, can be run from _PATH_ instead of specifying the exact location. It can easily discern the RPM-installed client from the one in a development folder.

Reply:
Removed.